src/Microsoft.Azure.WebJobs.Host/Common/ConverterManager.cs

```diff
@@ -0,0 +1,131 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.WebJobs
+{
+    // Concrete implementation of IConverterManager
+    internal class ConverterManager : IConverterManager
+    {
+        // Map from <TSrc,TDest> to a converter function. 
+        private Dictionary<string, object> _funcs = new Dictionary<string, object>();
+
+        public ConverterManager()
+        {
+            this.AddConverter<byte[], string>(DefaultByteArray2String);
+        }
+
+        private static string DefaultByteArray2String(byte[] bytes)
+        {
+            string str = Encoding.UTF8.GetString(bytes);
+            return str;
+        }
+
+        private static string GetKey<TSrc, TDest>()
+        {
+            return typeof(TSrc).FullName + "|" + typeof(TDest).FullName;
+        }
+
+        public void AddConverter<TSrc, TDest>(Func<TSrc, TDest> converter)
+        {
+            string key = GetKey<TSrc, TDest>();
```

Comment:
This is a good design conversation.  Normally, the bindings are disjoint since the src or dst should be a "native" type specific to my extensions (ie, EventData or CloudQueueMessage). 
In theory, this can help us make bindings compose too. IE, I can define a Stream-->WebImage and then multiple stream binders can leverage it. 


---
In reply to: [54931562](https://github.com/Azure/azure-webjobs-sdk/pull/643#discussion_r54931562) [](ancestors = 54931562)